### PR TITLE
Update Script WinEventParser #2

### PR DESCRIPTION
**Pull request by z0ne323**
- Adding WinEventParser + Renaming Repo
- Update Script WinEventParser #1
- Update Script WinEventParser #2

### CODE REVIEW

File: WinEventParser/main.py [hunk 1]
```diff
@@ -20,38 +20,38 @@ def admin():
         return False
 
 
+def is_valid_char(char):
+    """Function to check if each char of timestamp are valid"""
+    acceptable_chars = ['-', ' ', ':']
+    return char.isdigit() or char in acceptable_chars
+
+
+def get_chars_to_add(mode):
```
Comment (nigleweber) [inline WinEventParser/main.py:29]: Function is great!
Let's rename a bit so others can immediately understand what kind of characters are being added.

File: WinEventParser/main.py [hunk 2]
```diff
@@ -20,38 +20,38 @@ def admin():
...
-    timestamp_to_be_parsed = timestamp_to_be_parsed.split()
-    if len(timestamp_to_be_parsed) == 2:
-        timestamp_to_be_parsed = timestamp_to_be_parsed[0] + ' ' + timestamp_to_be_parsed[1]
-    else:
-        timestamp_to_be_parsed = timestamp_to_be_parsed[0]
-    acceptable_chars = ['-', ' ', ':']
-    for ttbp in timestamp_to_be_parsed:
-        if (ttbp.isdigit() or ttbp in acceptable_chars) is not True:
-            return None, None
+    timestamp_to_be_parsed = ' '.join(timestamp_to_be_parsed.split()[:2])
```
Comment (nigleweber) [inline WinEventParser/main.py:49]: Line 49 through 51 should be shifted to a new function as it has markedly different functionality than the rest of the function.
Line 47 and 48 should likely move to this new function as well.

File: WinEventParser/main.py [hunk 3]
```diff
@@ -20,38 +20,38 @@ def admin():
...
+    if not all(is_valid_char(c) for c in timestamp_to_be_parsed):
+        return None, None
     try:
```
Comment (nigleweber) [inline WinEventParser/main.py:52]: Try block is looking good overall

File: WinEventParser/main.py [hunk 4]
```diff
@@ -69,39 +69,50 @@ def convert_time_in_utc(time_local_timezone):
     return time_utc
 
 
+def get_num_of_timestamps(input_mode_timestamp):
```
Comment (nigleweber) [inline WinEventParser/main.py:72]: 👍

File: WinEventParser/main.py [hunk 5]
```diff
@@ -69,39 +69,50 @@ def convert_time_in_utc(time_local_timezone):
...
+    """Function used to get the number of timestamps the user wants to input"""
+    if input_mode_timestamp == "two":
+        return 2
+    return 1
+
+
+def get_input_mode(iteration_of_for_loop):
```
Comment (nigleweber) [inline WinEventParser/main.py:79]: 👍

File: WinEventParser/main.py [hunk 6]
```diff
@@ -69,39 +69,50 @@ def convert_time_in_utc(time_local_timezone):
...
+    """Function used to get the input mode for a timestamp"""
+    mode_to_pass = ""
+    mode_to_check = ["1", "2", "3", "4", "5", "6"]
+    while mode_to_pass not in mode_to_check:
+        mode_to_pass = input(f"[{iteration_of_for_loop}] Please when you input your timestamp do you want to input 1) a full timestamp, 2) a timestamp with date, hour and minute, 3) a timestamp with date and hour, 4) a timestamp with date only, 5) a timestamp with year and month, 6) a timestamp with year (1/2/3/4/5/6): ")
+    return int(mode_to_pass)
+
+
 def get_input_timestamp(input_mode_timestamp):
```
Comment (nigleweber) [inline WinEventParser/main.py:88]: Looking better. Still needs some refactoring.

Comment (z0ne323) [reply WinEventParser/main.py:88]: Yeah totally ! Just I was more focused on the first one you assigned but for sure I'll modify it ASAP

File: WinEventParser/main.py (hunk 1 above)
Comment (z0ne323) [reply WinEventParser/main.py:29]: something like get_string_to_add_to_timestamp() or get_completion_of_timestamp() ?

**Pull request closed without merging**